Add otelbot GitHub App to assets (#2548)
open-telemetry/community: assets.md

  * [Easy CLA](#easy-cla)
  * [Docker Hub](#docker-hub)

>   * [OpenTelemetry Bot](#opentelemetry-bot)
>   * [otelbot](#otelbot)

Reviewer: Given that you want folks to favor the `otelbot`, let's put that section ahead of the OTel bot (change it here and in the prose). WDYT?

```suggestion
  * [`otelbot`](#otelbot)
  * [OpenTelemetry Bot](#opentelemetry-bot)
```

2. When you use the built-in `secrets.GITHUB_TOKEN` to generate a pull request from inside a [GitHub Action], workflows
   will not run on that new pull request without closing and re-opening it manually (this limitation is in place to
   prevent accidental recursive workflow runs).

   The OpenTelemetry GitHub organization has a GitHub Action secret (`OTELBOT_PRIVATE_KEY`)
   and a GitHub Action variable `OTELBOT_APP_ID` that can be used to create a GitHub App token
   which will bypass this limitation, e.g.

   ```
   - uses: actions/create-github-app-token@v1
     id: app-token
     with:
       app-id: ${{ vars.OTELBOT_APP_ID }}
       private-key: ${{ secrets.OTELBOT_PRIVATE_KEY }}

   - name: Create pull request
     env:
       # not using secrets.GITHUB_TOKEN since pull requests from that token do not trigger workflows
       GH_TOKEN: ${{ steps.app-token.outputs.token }}
     run: ...
   ```

> [!WARNING]
> The `otelbot` is and needs to remain a **_public_ GitHub App** in order for EasyCLA to be able to verify its CLA status.

[GitHub Action]: https://docs.github.com/en/actions
[GitHub App]: https://docs.github.com/en/apps

### OpenTelemetry Bot

> > [!NOTE]
> > Consider using [otelbot](#otelbot) GitHub App listed below instead.

Reviewer: ```suggestion
> [!NOTE]
> Consider using the [otelbot](#otelbot) GitHub App instead.
```

   [Personal Access Token]: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token

> ### otelbot

Reviewer: ```suggestion
### `otelbot`
```

> This is a GitHub App owned by [@open-telemetry](https://github.com) that you can use when

Reviewer 2: ```suggestion
This is a GitHub App owned by [@open-telemetry](https://github.com/open-telemetry) that you can use when
```

like that?

automating common GitHub tasks in OpenTelemetry repos (e.g. release automation tasks).

This GitHub App has the following permissions:

- Read access to metadata
- Read and write access to pull requests

> 
> - Admins: [@open-telemetry/admins](https://github.com/orgs/open-telemetry/teams/admins

Author: (I think keeping the `Admins:` just for consistency with other sections)
```suggestion

- Admins: [@open-telemetry/admins](https://github.com/orgs/open-telemetry/teams/admins)
```

This GitHub App addresses two common issues:

1. Since you can't push directly to `main` from workflows (due to branch protections), the next best thing is to
   generate a pull request from the automation and use an account which has signed the CLA as the commit author.

   The OpenTelemetry Bot account has signed the CNCF CLA, and you can assign it as the commit author in your automation:

   ```
   git config user.name otelbot
   git config user.email 197425009+otelbot@users.noreply.github.com
   ```

   It is recommended to push to branch names that start with `otelbot/`, and to add a branch protection

>    rule for `otelbot/**/**` with the same setup as documented for

Reviewer: Just curious, why the stutter? Doesn't `**` glob everything and so `otelbox/**` would be enough, right?

>    [`dependabot/**/**`](docs/how-to-configure-new-repository.md#branch-protection-rule-dependabot). Note that branch protection rule ordering matters, so you will need to
>    delete the `**/**` branch protection rule temporarily, then add the `otelbot/**/**` branch protection
>    rule, then add back the `**/**` branch protection rule.

Reviewer: Callout the cautionary note:

```suggestion
   [`dependabot/**/**`](docs/how-to-configure-new-repository.md#branch-protection-rule-dependabot).
   
   > [!CAUTION]
   > Branch protection rule **ordering** matters, so you will need to delete the `**/**` branch protection rule temporarily, then add the `otelbot/**/**` branch protection rule, then add back the `**/**` branch protection rule.
```

Not sure if the above is properly formatted, but you get the idea.

> 2. When you use the built-in `secrets.GITHUB_TOKEN` to generate a pull request from inside a GitHub Action, workflows

Reviewer: ```suggestion
2. When you use the built-in `secrets.GITHUB_TOKEN` to generate a pull request from inside a [GitHub Action], workflows
```

   will not run on that new pull request without closing and re-opening it manually (this limitation is in place to

>    prevent accidental recursive workflow runs).

Reviewer: Side note: while I understand the restriction, I've been wondering ... do you think that there can be a better way than closing and opening a PR? For example, could we implement a `/kick` action that, say, the OTel bot could respond to by re-triggering / allowing GH actions over a PR? (I've tried force-pushing an empty commit, but that didn't seem to work.)

Author: sounds like a good thought, I don't think I've tried that

   The OpenTelemetry GitHub organization has a GitHub Action secret (`OTELBOT_PRIVATE_KEY`)
   and a GitHub Action variable `OTELBOT_APP_ID` that can be used to create a GitHub App token
   which will bypass this limitation, e.g.

   ```
   - uses: actions/create-github-app-token@v1
     id: app-token
     with:
       app-id: ${{ vars.OTELBOT_APP_ID }}
       private-key: ${{ secrets.OTELBOT_PRIVATE_KEY }}

   - name: Create pull request
     env:
       # not using secrets.GITHUB_TOKEN since pull requests from that token do not trigger workflows
       GH_TOKEN: ${{ steps.app-token.outputs.token }}
     run: ...
   ```

> [!WARNING]

> > The otelbot GitHub needs to remain a public GitHub App in order for EasyCLA to be able to verify its CLA status.

Reviewer: There seems to be a word missing?

```suggestion
> The `otelbot` is and needs to remain a **public GitHub App** in order for EasyCLA to be able to verify its CLA status.
```
